File: Algorithm/day2/ex00e5.py
q=len
def r(o, m, s):
    if q(s)>n: return 
    elif m>=k and q(s)==n:print(s)
    else:r(0,m,s+'0');r(o+1,max(m,o+1),s+'1')
n,k=map(int,input().split())
r(0,0,"")

File: Algorithm/day2/ex00e5.cpp
#include <bits/stdc++.h>
using namespace std;
int n, k;
void rec(int o, int m, string s) {
    if (s.size() > n) return;
    if (m >= k && s.size() == n) cout << s << '\n';
    rec(0, m, s + '0'); rec(o + 1, max(m, o + 1), s + '1');
}
int main() {
    cin >> n >> k; rec(0, 0, "");
}